# 03. Business Insights & Recommendations

This section summarizes the key findings from the exploratory data analysis and translates them into actionable business insights and recommendations.

## 1. Overall Business Performance

In [48]:
import pandas as pd

df = pd.read_csv("../Dataset/ecommerce_sales_dataset.csv")

df["Order_Date"] = pd.to_datetime(df["Order_Date"])

print("Data loaded successfully")
print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

Data loaded successfully
Rows: 10000
Columns: 26


In [49]:
print("Total Revenue:", round(df["Revenue"].sum(), 2))
print("Total Profit:", round(df["Profit"].sum(), 2))
print("Overall Profit Margin:", round((df["Profit"].sum() / df["Revenue"].sum()) * 100, 2), "%")
print("Total Orders:", df["Order_ID"].nunique())

Total Revenue: 5284387.7
Total Profit: 1437638.31
Overall Profit Margin: 27.21 %
Total Orders: 10000


## 2. Category Performance Insights

In [50]:
category_analysis = (
    df.groupby("Category")
    .agg(
        Orders=("Order_ID", "count"),
        Revenue=("Revenue", "sum"),
        Profit=("Profit", "sum")
    )
)

category_analysis["Profit_Margin_%"] = (
    category_analysis["Profit"] / category_analysis["Revenue"] * 100
).round(2)

category_analysis = category_analysis.sort_values("Revenue", ascending=False)

category_analysis

,Orders,Revenue,Profit,Profit_Margin_%
Category,,,,
Electronics,1977,3382028.46,925448.15,27.36
Home & Kitchen,2008,892842.02,232421.45,26.03
Clothing,1977,407471.79,115705.29,28.40
Books & Media,2043,386644.46,105037.69,27.17
Beauty & Health,1995,215400.97,59025.73,27.40


In [51]:
best_category_revenue = category_analysis["Revenue"].idxmax()
best_category_profit = category_analysis["Profit"].idxmax()
best_category_margin = category_analysis["Profit_Margin_%"].idxmax()

print("Highest Revenue Category:", best_category_revenue)
print("Highest Profit Category:", best_category_profit)
print("Highest Profit Margin Category:", best_category_margin)

Highest Revenue Category: Electronics
Highest Profit Category: Electronics
Highest Profit Margin Category: Clothing


## 3. Sub-Category Performance Insights

In [52]:
subcategory_analysis = (
    df.groupby("Sub_Category")
    .agg(
        Orders=("Order_ID", "count"),
        Revenue=("Revenue", "sum"),
        Profit=("Profit", "sum")
    )
)

subcategory_analysis["Profit_Margin_%"] = (
    subcategory_analysis["Profit"] / subcategory_analysis["Revenue"] * 100
).round(2)

subcategory_analysis = subcategory_analysis.sort_values(
    "Revenue", ascending=False
)

subcategory_analysis

,Orders,Revenue,Profit,Profit_Margin_%
Sub_Category,,,,
Laptops,486,1641700.18,446773.61,27.21
Smartphones,514,843506.83,227872.08,27.01
Tablets,498,688981.62,195392.04,28.36
Furniture,654,498779.83,125162.63,25.09
Software,1021,329315.22,89844.78,27.28
Appliances,683,272102.17,74022.00,27.20
Headphones,479,207839.83,55410.42,26.66
Footwear,642,180627.15,50856.61,28.16
Women's Wear,688,131973.90,38329.19,29.04


In [53]:
print("Highest Revenue Sub-Category:",
      subcategory_analysis["Revenue"].idxmax())

print("Highest Profit Sub-Category:",
      subcategory_analysis["Profit"].idxmax())

print("Highest Profit Margin Sub-Category:",
      subcategory_analysis["Profit_Margin_%"].idxmax())

Highest Revenue Sub-Category: Laptops
Highest Profit Sub-Category: Laptops
Highest Profit Margin Sub-Category: Women's Wear


## 4. Geographical & Regional Insights

In [54]:
country_analysis = (
    df.groupby("Country")
    .agg(
        Orders=("Order_ID", "count"),
        Revenue=("Revenue", "sum"),
        Profit=("Profit", "sum")
    )
)

country_analysis["Profit_Margin_%"] = (
    country_analysis["Profit"] / country_analysis["Revenue"] * 100
).round(2)

country_analysis = country_analysis.sort_values(
    "Revenue", ascending=False
)

country_analysis.head(10)

,Orders,Revenue,Profit,Profit_Margin_%
Country,,,,
Canada,876,454911.29,121002.80,26.60
USA,829,444541.88,123170.72,27.71
Mexico,897,431676.58,122543.90,28.39
Egypt,481,344299.35,103930.99,30.19
Jordan,509,297916.85,85073.63,28.56
China,516,295780.41,73762.21,24.94
Japan,484,288011.04,74848.58,25.99
Italy,512,279837.36,77982.55,27.87
India,523,271854.16,79751.95,29.34


In [55]:
region_analysis = (
    df.groupby("Region")
    .agg(
        Orders=("Order_ID", "count"),
        Revenue=("Revenue", "sum"),
        Profit=("Profit", "sum")
    )
)

region_analysis["Profit_Margin_%"] = (
    region_analysis["Profit"] / region_analysis["Revenue"] * 100
).round(2)

region_analysis = region_analysis.sort_values(
    "Revenue", ascending=False
)

region_analysis

,Orders,Revenue,Profit,Profit_Margin_%
Region,,,,
Middle East,2458,1348593.22,384522.85,28.51
North America,2602,1331129.75,366717.42,27.55
Asia,2494,1330007.18,350413.48,26.35
Europe,2446,1274657.55,335984.56,26.36


## 5. Customer Insights

In [56]:
segment_analysis = (
    df.groupby("Customer_Segment")
    .agg(
        Orders=("Order_ID", "count"),
        Revenue=("Revenue", "sum"),
        Profit=("Profit", "sum")
    )
)

segment_analysis["Profit_Margin_%"] = (
    segment_analysis["Profit"] / segment_analysis["Revenue"] * 100
).round(2)

segment_analysis = segment_analysis.sort_values(
    "Revenue", ascending=False
)

segment_analysis

,Orders,Revenue,Profit,Profit_Margin_%
Customer_Segment,,,,
Regular,5073,2717416.06,731759.88,26.93
New,1942,1017899.86,288703.24,28.36
Premium,1961,974664.57,273226.31,28.03
VIP,1024,574407.21,143948.88,25.06


In [57]:
gender_analysis = (
    df.groupby("Customer_Gender")
    .agg(
        Orders=("Order_ID", "count"),
        Revenue=("Revenue", "sum"),
        Profit=("Profit", "sum")
    )
)

gender_analysis["Profit_Margin_%"] = (
    gender_analysis["Profit"] / gender_analysis["Revenue"] * 100
).round(2)

gender_analysis

,Orders,Revenue,Profit,Profit_Margin_%
Customer_Gender,,,,
Female,5097,2660403.85,731186.48,27.48
Male,4903,2623983.85,706451.83,26.92


## 6. Order & Payment Insights

In [58]:
order_status_analysis = (
    df.groupby("Order_Status")
    .agg(
        Orders=("Order_ID", "count"),
        Revenue=("Revenue", "sum"),
        Profit=("Profit", "sum")
    )
)

order_status_analysis["Profit_Margin_%"] = (
    order_status_analysis["Profit"] / order_status_analysis["Revenue"] * 100
).round(2)

order_status_analysis

,Orders,Revenue,Profit,Profit_Margin_%
Order_Status,,,,
Cancelled,908,499132.22,123283.92,24.70
Delivered,6273,3308169.75,909034.36,27.48
Processing,962,537441.33,150815.96,28.06
Returned,1857,939644.40,254504.07,27.09


In [59]:
payment_analysis = (
    df.groupby("Payment_Method")
    .agg(
        Orders=("Order_ID", "count"),
        Revenue=("Revenue", "sum"),
        Profit=("Profit", "sum")
    )
)

payment_analysis["Profit_Margin_%"] = (
    payment_analysis["Profit"] / payment_analysis["Revenue"] * 100
).round(2)

payment_analysis = payment_analysis.sort_values(
    "Revenue", ascending=False
)

payment_analysis

,Orders,Revenue,Profit,Profit_Margin_%
Payment_Method,,,,
PayPal,1470,831620.67,220318.94,26.49
Debit Card,1486,791394.26,219462.41,27.73
Credit Card,1430,777949.51,205563.87,26.42
Google Pay,1422,771676.04,214943.29,27.85
Bank Transfer,1426,723569.44,196679.30,27.18
Cash on Delivery,1414,722989.12,203581.51,28.16
Apple Pay,1352,665188.66,177088.99,26.62


## 7. Key Business Findings

### Key Findings

- Electronics is the highest revenue and profit-generating category.
- Clothing has the highest profit margin among categories.
- Laptops are the strongest sub-category in terms of revenue and profit.
- Women's Wear has the highest profit margin among sub-categories.
- Middle East is the highest-revenue region and also has the highest regional profit.
- Regular customers contribute the highest revenue and profit.
- Female customers generate slightly higher revenue and profit than male customers.
- Delivered orders contribute the majority of orders, revenue, and profit.
- PayPal generates the highest revenue among payment methods, while Cash on Delivery has the highest profit margin.

## 8. Business Recommendations

### Recommendations

- Focus on Electronics and Laptops to sustain the strongest revenue and profit contribution.
- Explore opportunities to increase the contribution of high-margin categories such as Clothing.
- Strengthen marketing and retention strategies for Regular customers while developing New and Premium customers.
- Investigate the lower profit margin of VIP customers and identify opportunities to improve their profitability.
- Prioritize growth opportunities in the Middle East while maintaining performance across other regions.
- Monitor cancelled and returned orders to reduce potential revenue and profitability losses.
- Continue supporting multiple payment methods while evaluating high-margin methods such as Cash on Delivery and Google Pay.

## 9. Conclusion

The analysis shows a healthy overall business performance with total revenue of 5.28M, total profit of 1.44M, and an overall profit margin of 27.21%.

Electronics and Laptops are the strongest revenue and profit contributors, while Clothing and Women's Wear show strong profit margins. The Middle East is the leading region by revenue and profit, while Regular customers contribute the largest share of customer revenue and profit.

The business can improve further by focusing on high-performing products and regions, strengthening customer retention, improving VIP customer profitability, and reducing cancelled and returned orders.